actionpack/test/controller/parameters/nested_parameters_permit_test.rb

```diff
@@ -150,6 +150,29 @@ def assert_filtered_out(params, key)
     assert_filtered_out permitted[:book][:authors_attributes]["0"], :age_of_death
   end
 
+  test "fields_for-style nested params with non_numeric keys" do
+    params = ActionController::Parameters.new(
+      book: {
+        authors_attributes: {
+          '0': { name: "William Shakespeare", age_of_death: "52" },
+          '1': { name: "Unattributed Assistant" },
+          'new_record': { name: "Some name" }
```

Comment:
You should add a `'2': "NOT A HASH"` nested attribute to this test to make sure that case gets filtered too.